src/Testcontainers.CosmosDb/CosmosDbBuilder.cs

```diff
@@ -0,0 +1,121 @@
+namespace Testcontainers.CosmosDb;
+
+/// <inheritdoc cref="ContainerBuilder{TBuilderEntity, TContainerEntity, TConfigurationEntity}" />
+[PublicAPI]
+public sealed class CosmosDbBuilder : ContainerBuilder<CosmosDbBuilder, CosmosDbContainer, CosmosDbConfiguration>
+{
+    public const string CosmosDbImage = "mcr.microsoft.com/cosmosdb/linux/azure-cosmos-emulator:latest";
+
+    public const ushort CosmosDbPort = 8081;
+
+    public const string DefaultAccountKey = "C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==";
+
+    public const int DefaultPartitionCount = 2;
+
+    public const string DefaultIpAddressOverride = "127.0.0.1";
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CosmosDbBuilder" /> class.
+    /// </summary>
+    public CosmosDbBuilder()
+        : this(new CosmosDbConfiguration())
+    {
+        DockerResourceConfiguration = Init().DockerResourceConfiguration;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CosmosDbBuilder" /> class.
+    /// </summary>
+    /// <param name="resourceConfiguration">The Docker resource configuration.</param>
+    private CosmosDbBuilder(CosmosDbConfiguration resourceConfiguration)
+        : base(resourceConfiguration)
+    {
+        DockerResourceConfiguration = resourceConfiguration;
+    }
+
+    /// <inheritdoc />
+    protected override CosmosDbConfiguration DockerResourceConfiguration { get; }
+
+    /// <summary>
+    /// Sets the partition count.
+    /// </summary>
+    /// <param name="partitionCount">The number of partitions.</param>
+    /// <returns>A configured instance of <see cref="CosmosDbBuilder" />.</returns>
+    public CosmosDbBuilder WithPartitionCount(int partitionCount)
+    {
+        return Merge(DockerResourceConfiguration, new CosmosDbConfiguration(partitionCount: partitionCount))
+            .WithEnvironment("AZURE_COSMOS_EMULATOR_PARTITION_COUNT", partitionCount.ToString());
+    }
+
+    /// <summary>
+    /// Sets the overridden IP address.
+    /// </summary>
+    /// <param name="ipAddress">The overridden IP address.</param>
+    /// <returns>A configured instance of <see cref="CosmosDbBuilder" />.</returns>
+    public CosmosDbBuilder WithIpAddressOverride(string ipAddress)
+    {
+        return Merge(DockerResourceConfiguration, new CosmosDbConfiguration(ipAddressOverride: ipAddress))
+            .WithEnvironment("AZURE_COSMOS_EMULATOR_IP_ADDRESS_OVERRIDE", ipAddress);
+    }
```

Comment:
Do we really need these properties? They are never used except in the `Validate()` member. Normally, we do not provide APIs for simple environment variables. Instead, developers can set them using 'WithEnvironment(k, v)'. It can be difficult to determine which environment variables are worth an API and which are not. We generally avoid creating a dedicated API if the value is only needed as an environment variable.

Reply:
Ah OK, probably not. I just migrated the important properties from the obsolete version. I can remove them.

Reply:
I've removed these now 👍